### PR TITLE
Fixes #2753

### LINKED ISSUE #2753: Setting a timeout causes assertion failures in httpclient
``` nim
import httpclient

echo getContent("https://gitlab.3dicc.com/gokr/lapp", timeout=400)
```

```
Traceback (most recent call last)
a21.nim(3)               a21
httpclient.nim(491)      getContent
httpclient.nim(472)      get
httpclient.nim(445)      request
httpclient.nim(435)      request
httpclient.nim(245)      parseResponse
httpclient.nim(170)      parseBody
httpclient.nim(160)      parseChunks
net.nim(714)             recv
system.nim(3058)         failedAssertImpl
system.nim(2280)         raiseAssert
Error: unhandled exception: avail <= size - read  [AssertionError]
```

### PR DESCRIPTION
**Pull request by data-man**
Fixed as suggested by @skilchen.

### CODE REVIEW

**Comment by Araq** [comment]
No test case?

**Comment by data-man** [comment]
Added test from #2753

File: tests/stdlib/thttpclient.nim [hunk 1]
```diff
@@ -154,8 +154,20 @@ proc ipv6Test() =
...
+proc longTimeoutTest() =
+# Issue #2753
+  try:
+    var client = newHttpClient(timeout = 1000)
+    var resp = client.request("https://au.yahoo.com")
+    client.close()
+  except AssertionError:
+    doAssert false, "Exceptions should not be raised"
+  except:
+    discard
```
Comment (Araq) [inline tests/stdlib/thttpclient.nim]: What kind of test is that? "Fine as long as we get no AssertionError?!"

Comment (data-man) [reply tests/stdlib/thttpclient.nim]: Without this PR  ```doAssert``` runs.
Try exec thttpclient without this fix.

Comment (Araq) [reply tests/stdlib/thttpclient.nim]: Not my point. The test should check that either a TimeoutError is raised or nothing.

Comment (data-man) [reply tests/stdlib/thttpclient.nim]: Without try/except/this fix (and with #7813 :-)) new test outputs:
```
thttpclient.nim(177)     thttpclient
thttpclient.nim(171)     longTimeoutTest
httpclient.nim(1218)     request
httpclient.nim(1195)     request
httpclient.nim(1180)     requestAux
httpclient.nim(1073)     parseResponse
httpclient.nim(975)      parseBody
httpclient.nim(951)      parseChunks
httpclient.nim(910)      recvFull
net.nim(1221)            recv
net.nim(1199)            recv
net.nim(1170)            recv
system.nim(3846)         failedAssertImpl
system.nim(3839)         raiseAssert
system.nim(2849)         sysFatal
Error: unhandled exception: avail <= size - read  [AssertionError]
``

Comment (data-man) [reply tests/stdlib/thttpclient.nim]: Oh! In release mode sometimes:
```Error: unhandled exception: Server terminated connection prematurely [ProtocolError]```

sometimes:
```SIGSEGV: Illegal storage access. (Attempt to read from nil?)
Segmentation fault (core dumped)```

Comment (Araq) [reply tests/stdlib/thttpclient.nim]: Well I won't add wrong tests or tests that don't work yet.

**Comment by dom96** [comment]
@skilchen said that this is a "quick fix". To me it looks like a hack (sorry @skilchen).

**Comment by Araq** [comment]
> IMO this can just be merged without the test. Testing against live websites is too unreliable.

Ok, fine with me, so remove the test again...

**Pull request merged**